test/smoke/java/smoke-tests/src/main/java/org/kie/kogito/operator/tests/ops/BasicOpenshiftOperations.java

```diff
@@ -0,0 +1,198 @@
+    }
+
+    public void loadAdminResource(Path path) {
+        loadResource(openShiftAdmin, path);
+    }
+
+    public URL getRoute(String serviceName) {
+        return getHttpsRouteUrl(serviceName)
+                .orElse(getHttpRouteUrl(serviceName)
+                        .orElseThrow(() -> new RuntimeException("No route found for service " + serviceName)));
+    }
+
+    public Response executeHttpRequest(Request request) {
+        LOGGER.info("Load http request {}", request);
+        OkHttpClient client = getMaster().getHttpClient();
+        try {
+            return client.newCall(request).execute();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void loadResource(OpenShift openShift, Path path) {
+        LOGGER.info("Load path {}", path.toString());
+        try (InputStream inputStream = Files.newInputStream(path)) {
+            openShift.load(inputStream).deletingExisting().createOrReplace();
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot load resource path " + path, e);
+        }
+    }
+
+    protected Optional<URL> getHttpRouteUrl(String serviceName) {
+        return getRouteUri(Protocol.http, serviceName);
+    }
+
+    protected Optional<URL> getHttpsRouteUrl(String serviceName) {
+        return getRouteUri(Protocol.https, serviceName);
+    }
+
+    private Optional<URL> getRouteUri(Protocol protocol, String serviceName) {
+        Service service = openShift.getService(serviceName);
+        if (service == null) {
+            throw new RuntimeException("No service with name " + serviceName + " found ...");
+        }
+        Predicate<Route> httpsPredicate = n -> n.getSpec().getTls() != null;
+        Predicate<Route> httpPredicate = n -> n.getSpec().getTls() == null;
+
+        String routeHost = null;
+        List<Route> routes = openShift.getRoutes();
+        Optional<Route> route = routes.stream()
+                .filter(protocol == Protocol.https ? httpsPredicate : httpPredicate)
+                .filter(n -> n.getSpec().getTo().getName().equals(serviceName))
+                .findAny();
+        if (route.isPresent()) {
+            routeHost = route.get().getSpec().getHost();
+        } else {
+            String routeNames = routes.stream()
+                    .map(n -> n.getMetadata().getName())
+                    .collect(Collectors.joining(", "));
+            LOGGER.warn(protocol + " route leading to service " + serviceName + " not found. Available routes " + routeNames);
+            return Optional.empty();
+        }
+        String uriValue = protocol.name() + "://" + routeHost + ":" + retrievePort(protocol);
+
+        try {
+            return Optional.of(new URI(uriValue.toString()).toURL());
+        } catch (URISyntaxException | MalformedURLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String retrievePort(Protocol protocol) {
```

Comment:
wouldn't be better create a different class and put the enum plus this method there?

Reply:
Updated the enum